### PR TITLE
[CI] Add Prepare Release and VS Insertion stages

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/yaml-templates/pull/117

Updates the .NET 6 NuGet packaging steps to exclude package metadata,
as the .msi conversion tooling does not process .nupkg file names with
the `+sha.commit` metadata.

Two new stages have been added to facilitate the Visual Studio setup
authoring process.

The first stage named "Prepare Release" will sign the .NET 6 NuGet
package content (inside and out), convert relevant packages to .msi
installers, generate Visual Studio manifests for the .msi installers,
and push the signed packages to the `xamarin-impl` feed.

The new `SignList.xml` file is required for our NuGet signing templates.

The new `xamarin-workload.props` file contains version information
and other metadata required to generate a Visual Studio manifest.

The second stage starts with a [manual validation task][0].  This task
will pause and wait for someone to click a "Resume" or "Reject" button
that will appear on the pipeline UI.  This task is configured to be
rejected after waiting for two days, but it can be manually re-ran at a
later date if we want to trigger VS insertion for an older build.

If the manual validation task is approved, a VS Drop will be created
containing all .NET 6 .msi files.  This Drop URL can then be used to
update our component versions in Visual Studio.  This last piece is
currently manual as we will initially be introducing new components,
however we should be able to automate VS PR creation in the future.

[0]: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/manual-validation?view=azure-devops&tabs=yaml

### CODE REVIEW

File: dotnet/Makefile
Comment (pjcollins) [inline dotnet/Makefile:165]: @rolfbjarne did something else require that the package file names have the `+sha.commit` metadata in them? Or was this done for convenience? The .msi conversion tooling fails to process these package names, but we could potentially change something over there if this is needed.

File: tools/devops/automation/build-pipeline.yml
Comment (pjcollins) [inline tools/devops/automation/build-pipeline.yml:266]: @rolfbjarne @Redth @jonathanpeppers Do you have thoughts on which maciOS workloads should we include in VS Windows? Right now this tooling is picking up iOS, tvOS, and Mac Catalyst. I was thinking we should at least insert everything that MAUI depends on, perhaps we should only insert iOS and Mac Catalyst on Windows?

Comment (jonathanpeppers) [reply tools/devops/automation/build-pipeline.yml:266]: I would think it would probably be fine to not insert the macOS packs in VS? What about tvOS?

A dev could still *get* the packs from `dotnet workload install` if needed, but I don't think we need a checkbox & component to show up in the VS installer.

So far, I've only heard about "remote build" support for iOS & MacCatalyst, and MAUI only uses these two workloads so far:

* https://github.com/dotnet/maui/blob/c3083b9d9d229ff221483a03003eb5bd53692503/src/Workload/Microsoft.NET.Sdk.Maui/WorkloadManifest.in.json#L53-L56
* https://github.com/dotnet/maui/blob/c3083b9d9d229ff221483a03003eb5bd53692503/src/Workload/Microsoft.NET.Sdk.Maui/WorkloadManifest.in.json#L68-L71

Comment (rolfbjarne) [reply tools/devops/automation/build-pipeline.yml:266]: I'm pretty sure we don't have any support for `macOS` in VS Windows (so it wouldn't make sense to add them to the installer).

I'm not sure we have tvOS support either, but I've never heard anything either way.

@emaf can confirm though

File: dotnet/Makefile
Comment (rolfbjarne) [reply dotnet/Makefile:165]: IIRC it was just convenience, so it should be good.

File: dotnet/Makefile
Comment (rolfbjarne) [inline dotnet/Makefile:137]: Something like this should work, and keep working when/if new platforms are added:

```suggestion
		$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),-e "s/@$(platform)_WORKLOAD_VERSION@/$($(platform)_NUGET_VERSION).$($(platform)_NUGET_COMMIT_DISTANCE)/g)" \
```

File: dotnet/Workloads/xamarin-workload.template.props
Comment (pjcollins) [inline dotnet/Workloads/xamarin-workload.template.props:13]: The `Title` and `Description` metadata will show up in the VS installer, all of this quick naming should be improved and approved.

**Comment by pjcollins** [comment]
@rolfbjarne @mandel-macaque aside from the `Publish Nugets to xamarin-impl` step being disabled for now, this is ready for review. I will test the changes to feed pushing last once everything else is approved. 

There is still work pending in dotnet/runtime to unblock VS windows insertion of the .msis that this PR will now generate. However, I think we are ready to merge this in its current form to get the NuGet signing and .msi creation changes in sooner than later.

**Review by rolfbjarne — APPROVED** [review]

File: tools/devops/automation/templates/build/publish-nugets.yml
Comment (jonathanpeppers) [inline tools/devops/automation/templates/build/publish-nugets.yml:15]: Will this prevent dotnet/maui from consuming xamarin-macios builds?

I see where the other `nuget push` command is turned off for now.

Comment (pjcollins) [reply tools/devops/automation/templates/build/publish-nugets.yml:15]: I mentioned this briefly above but its easy to lose comments in the build/test reporting. I had planned on testing this change last once the rest of the yaml and build changes were approved.

File: tools/devops/automation/build-pipeline.yml
Comment (mandel-macaque) [inline tools/devops/automation/build-pipeline.yml:303]: Can we move this to a template like with the others to try and make this file as simple as possible. Perse I have no problem but I'd like not to have to look at 30000 lines of yaml :)

**Review by mandel-macaque — APPROVED** [review]

**Comment by pjcollins** [comment]
@mandel-macaque @rolfbjarne I think this is ready to go, feed uploading is working with the latest commit: https://dev.azure.com/azure-public/vside/_packaging?_a=package&feed=xamarin-impl&package=Microsoft.iOS.Runtime.iossimulator-x64&protocolType=NuGet&version=15.0.100-ci.pjc-nuget-convert-insert.221%2Bsha.03d0fc7be&view=overview. I am unable to merge myself since this has drifted from main since last night, let me know if anything else needs to change. I think we can squash and merge this one using the description above, there were lots of yaml formatting/syntax changes that don't need mentioning.

**Comment by rolfbjarne** [comment]
Test failures are unrelated


* monotouch-test/Mac [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2469
* dont link/Mac Catalyst/Debug: https://github.com/xamarin/maccore/issues/2414

**Review by rolfbjarne — APPROVED** [review]

**Comment by mandel-macaque** [comment]
This broke PR builds :/

**Comment by pjcollins** [comment]
Sorry, I missed a condition on one of the MicroBuild signing plugin installation steps... https://github.com/xamarin/xamarin-macios/pull/12157/commits/c2f9bc09888ac9a7a8b38992a4f938ab3e5f5648. We can restore this in https://github.com/xamarin/xamarin-macios/pull/12157

**Pull request merged**
